include/net/softap_wifi_provision.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file softap_wifi_provision.h
+ * @brief Header file for the softAP Wi-Fi Provision Library.
+ */
+
+#ifndef SOFTAP_WIFI_PROVISION_H__
+#define SOFTAP_WIFI_PROVISION_H__
+
+/**
+ * @defgroup softap_wifi_provision_library SoftAP Wi-Fi Provision library
+ * @{
+ * @brief Library used to provision a Wi-Fi device to a Wi-Fi network using HTTPS in softAP mode.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Types of events generated during provisioning.
+ */
+enum softap_wifi_provision_evt_type {
+	/** The provisioning process has started. */
+	SOFTAP_WIFI_PROVISION_EVT_STARTED,
+
+	/** A client (station) has connected to the softAP Wi-Fi network. */
+	SOFTAP_WIFI_PROVISION_EVT_CLIENT_CONNECTED,
+
+	/** A client (station) has disconnected from the softAP Wi-Fi network. */
+	SOFTAP_WIFI_PROVISION_EVT_CLIENT_DISCONNECTED,
+
+	/** Wi-Fi credentials received from the client. */
+	SOFTAP_WIFI_PROVISION_EVT_CREDENTIALS_RECEIVED,
+
+	/** The provisioning process has completed. */
+	SOFTAP_WIFI_PROVISION_EVT_COMPLETED,
+
+	/** Wi-Fi credentials deleted, a reboot is required to enter an unprovisioned state. */
+	SOFTAP_WIFI_PROVISION_EVT_RESET_REBOOT_REQUEST,
```

Comment:
Credentials-deleted is not reflected in the enum's name (neither does the resulting state: ie. the device is becoming un-provisioned). Reboot request is essentially the outcome of the device becoming un-provisioned and could change in the future. Again, that's reflecting only the enum naming, so not so important.

Reply:
Ill see if I can find a bit more suiting name.